acertmgr/authority/v2.py

```diff
@@ -260,8 +260,8 @@ def get_crt_from_csr(self, csr, domains, challenge_handlers):
         if code >= 400:
             raise ValueError("Error downloading certificate chain: {0} {1}".format(code, certificate))
 
-        cert_dict = re.match((r'(?P<cert>-----BEGIN CERTIFICATE-----[^\-]+-----END CERTIFICATE-----)\n\n'
-                              r'(?P<ca>-----BEGIN CERTIFICATE-----[^\-]+-----END CERTIFICATE-----)?'),
+        cert_dict = re.match((r'(?P<cert>^-----BEGIN CERTIFICATE-----\n[^\-]+\n-----END CERTIFICATE-----)\n*'
+                              r'(?P<ca>-----BEGIN CERTIFICATE-----\n.+\n-----END CERTIFICATE-----)?$'),
```

Comment:
I don't understand this change: should intermediate certificates not be part of the "ca" section? If it does not matter, why does this not return a simple list under a single name?

Reply:
According to RFC the first certificate is the endpoint certifcate (cert), the second one is the intermediate ca certifacte and all following certificates are the rest of the CA chain (all put into ca). This regex just splits the Certificate an CA Chain into separate chains for processing with tools.convert_pem_str_to_cert(...).